harness/determined/util.py

```diff
@@ -20,6 +20,11 @@
 
 @util.preserve_random_state
 def download_gcs_blob_with_backoff(blob: Any, n_retries: int = 32, max_backoff: int = 32) -> Any:
+    from google.cloud.storage import Blob
+
+    assert isinstance(
+        blob, Blob
+    ), f"Called download_gcs_blob_with_backoff with object of type {type(blob)}"
```

Comment:
nit: can you use `type(blob).__name__`?

Reply:
Can you say more about why that's preferable?